tests/test_phase5_rebuild_ete.py

```
@@ -0,0 +1,785 @@
+    # Frame 2 (50 ms later): ~31 m before T4 at 142 km/h → BRAKE NOW
+    # This MUST NOT be collapsed by the (kind=brake, cornerLabel=T4) key
+    opts2 = lua.eval(
+        "{ splinePos = 0.443, currentSpeedKmh = 142, dt = 0.05, "
+        "  bestSortedTrace = nil, brakingPoints = nil, segments = nil, trackLengthM = 4500 }"
+    )
+    opts2["bestSortedTrace"] = trace
+    opts2["brakingPoints"] = brakes
+    opts2["segments"] = segments
+    view2 = rtc["tick"](opts2)
+    primary2 = (view2["primaryLine"] or "").upper()
+    assert "BRAKE NOW" in primary2 or "BRAKE" == primary2.split()[0], (
```

Ruff (0.15.9)

[warning] Yoda condition detected

Rewrite as `primary2.split()[0] == "BRAKE"`

(SIM300)